code/datums/components/storage/storage.dm

```diff
@@ -57,6 +57,10 @@
 	var/screen_start_x = 4								//These two are where the storage starts being rendered, screen_loc wise.
 	var/screen_start_y = 2
 	//End
+	
+	var/limited_random_access = FALSE					//Quick if statement in accessible_items to determine if we care at all about what people can access at once.
+	var/limited_random_access_stack_position = 0					//If >0, can only access top <x> items
+	var/limited_random_access_stack_bottom_up = FALSE				//If TRUE, above becomes bottom <x> items

```

Comment:
Can't you make these var names any shorter?
It's kinda bugging me. I mean, just think how obnoxious it would be if you merged lines 157-159 in a single line right now.
Otherwise, cool.

Reply:
@Ghommie the idea is in the future you may have more checks, i made the varnames verbose because random access is different from things like dumping out where you're just sequentially taking everything
the boolean is to skip over all other checks for if in the future the checks are more processor heavy

Reply:
I mean, you can rename limited_random_access_stack_position to limited_random_access, delete the former and replace line 158 - 159 with:
`	if(random_access && limited_random_access && lengh(contents) > limited_random_access)`

And it'd still work just as fine. Save the complexity for future tinkering.